lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -165,25 +217,245 @@ namespace Js
         return tempAllocatorObject->GetAllocator();
     }
 
-    bool SourceTextModuleRecord::ResolveExport(PropertyId exportName, ResolutionDictionary* resolveSet, ResolveSet* exportStarSet, ModuleNameRecord** exportRecord)
+    ExportedNames* SourceTextModuleRecord::GetExportedNames(ExportModuleRecordList* exportStarSet)
     {
+        if (exportedNames != nullptr)
+        {
+            return exportedNames;
+        }
         ArenaAllocator* allocator = EnsureTempAllocator();
-        if (resolveSet == nullptr)
+        if (exportStarSet->Has(this))
+        {
+            return nullptr;
+        }
+        exportStarSet->Prepend(this);
+        ExportedNames* localNames = nullptr;
+        if (this->localExportRecordList != nullptr)
+        {
+            localNames = (ExportedNames*)AllocatorNew(ArenaAllocator, allocator, ExportedNames, allocator);
+            this->localExportRecordList->Map([=](ModuleExportEntry exportEntry) {
+                PropertyId exportNameId = EnsurePropertyIdForIdentifier(exportEntry.exportName);
+                localNames->Prepend(exportNameId);
+            });
+        }
+        if (this->indirectExportRecordList != nullptr)
+        {
+            if (localNames == nullptr)
+            {
+                localNames = (ExportedNames*)AllocatorNew(ArenaAllocator, allocator, ExportedNames, allocator);
+            }
+            this->indirectExportRecordList->Map([=](ModuleExportEntry exportEntry) {
+                PropertyId exportedNameId = EnsurePropertyIdForIdentifier(exportEntry.exportName);
+                localNames->Prepend(exportedNameId);
+            });
+        }
```

Comment:
Do you feel strongly about merging these three lists of exports (indirectExportRecordList, localExportRecordList, starExportRecordList) ? Should be possible to determine which type of export without keeping them in different lists if duplicating this code becomes troublesome.